### PR TITLE
Pipelines v2 - Apple

### PR DESCRIPTION
**Pull request by mre**
This is based on the architecture of the Apple pipeline with the
same goals:

- Fully typed
- Decoupling responsibilities to add unit tests
- Make code scale vertically and use job queue
- Make code more idiomatic

### CODE REVIEW

File: apple/job/dates.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        # Calculate the number of days between start and end date
+        self.days = (end - start).days
+
+    def __iter__(self):
+        """
+        Iterate over all days in the range
+        """
+        for i in range(self.days):
+            yield (self.start - dt.timedelta(days=i), self.end - dt.timedelta(days=i))
```
Comment (woolfg) [inline apple/job/dates.py:23]: looks weird, see https://github.com/openpodcast/pipelines/pull/40#discussion_r1130818717

**Review by woolfg — APPROVED** [review]
looks good, we should do a test by executing the old code and new code and compare the database content to see if everything works as expected

File: apple/job/dates.py (hunk 1 above)
Comment (mre) [reply apple/job/dates.py:23]: true. fixing

Comment (mre) [reply apple/job/dates.py]: done

File: spotify/Makefile [hunk 2]
```diff
@@ -2,6 +2,10 @@
 run dev:
 	python3 main.py
 
+.PHONY: test
+test:
+	python3 -m pytest
+
```
Comment (mre) [inline spotify/Makefile:8]: I want to run unit tests with the pipelines now.
To keep things simple I decided to add this test target here, even tough it is unrelated to the rest of the changes

**Review by woolfg — APPROVED** [review]

**Comment by mre** [comment]
Local tests all pass. 🎉 Merging.

**Pull request merged**